### PR TITLE
Heavy cleanup of the draw_square tutorial.

### PR DESCRIPTION
**Pull request by clalancette**
In particular:
1. Make it conform to the current ROS 2 style.
2. Add in copyright information.
3. Refactor the entire code into a class, which tidies it up quite a bit and removes a bunch of globals.
4. Make sure to wait for the reset to complete before trying to move the turtle.

### CODE REVIEW

File: turtlesim/tutorials/draw_square.cpp [hunk 1]
```diff
@@ -1,168 +1,204 @@
...
-    commandTurtle(twist_pub, 0, 0);
+    current_pose_ = pose;
+    first_pose_set_ = true;
   }
-  else
+
+  bool hasReachedGoal()
   {
-    commandTurtle(twist_pub, 1.0f, 0);
+    return fabsf(current_pose_.x - goal_pose_.x) < 0.1 && fabsf(current_pose_.y - goal_pose_.y) < 0.1 && fabsf(current_pose_.theta - goal_pose_.theta) < 0.01;
```
Comment (ahcorde) [inline turtlesim/tutorials/draw_square.cpp]: line < 100

File: turtlesim/tutorials/draw_square.cpp [hunk 2]
```diff
@@ -1,168 +1,204 @@
...
-bool hasStopped()
-{
-  return g_pose.angular_velocity < 0.0001 && g_pose.linear_velocity < 0.0001;
-}
-
-void printGoal()
-{
-  RCLCPP_INFO(rclcpp::get_logger("draw_square"), "New goal [%f %f, %f]", g_goal.x, g_goal.y, g_goal.theta);
-}
+    pose_sub_ = this->create_subscription<turtlesim::msg::Pose>("turtle1/pose", 1, std::bind(&DrawSquare::poseCallback, this, std::placeholders::_1));
```
Comment (ahcorde) [inline turtlesim/tutorials/draw_square.cpp]: line < 100

**Review by ahcorde — CHANGES_REQUESTED** [review]
Does it make sense to include `ament_lint_auto` ?

**Comment by clalancette** [comment]
> Does it make sense to include ament_lint_auto ?

It's a good question.  I ended up not doing this because I would then have to change the style of all of the files in this package, and I wanted to concentrate on fixing `draw_square` for now.  I think we should eventually clean this up and do that, but as a separate PR.

That said, I'll do the updates you suggest above.

File: turtlesim/tutorials/draw_square.cpp (hunk 1 above)
Comment (clalancette) [reply turtlesim/tutorials/draw_square.cpp]: Fixed in 2e574e3

File: turtlesim/tutorials/draw_square.cpp (hunk 2 above)
Comment (clalancette) [reply turtlesim/tutorials/draw_square.cpp]: Fixed in 2e574e3

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18781/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13306/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19523/)

**Review by ahcorde — APPROVED** [review]

**Comment by gavanderhoorn** [comment]
@clalancette: could this potentially be backported to `humble`?

Without these changes, it looks like `draw_square` is broken on Humble.

**Pull request merged**